cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/ConnectBackgroundJobs.scala

```diff
@@ -2,49 +2,67 @@ package org.hyperledger.identus.agent.server.jobs
 
 import org.hyperledger.identus.agent.server.config.AppConfig
 import org.hyperledger.identus.agent.server.jobs.BackgroundJobError.ErrorResponseReceivedFromPeerAgent
-import org.hyperledger.identus.agent.walletapi.model.error.DIDSecretStorageError
-import org.hyperledger.identus.agent.walletapi.model.error.DIDSecretStorageError.{KeyNotFoundError, WalletNotFoundError}
 import org.hyperledger.identus.agent.walletapi.service.ManagedDIDService
 import org.hyperledger.identus.agent.walletapi.storage.DIDNonSecretStorage
-import org.hyperledger.identus.connect.core.model.error.ConnectionServiceError.{
-  InvalidStateForOperation,
-  RecordIdNotFound
-}
 import org.hyperledger.identus.connect.core.model.ConnectionRecord
 import org.hyperledger.identus.connect.core.model.ConnectionRecord.*
 import org.hyperledger.identus.connect.core.service.ConnectionService
 import org.hyperledger.identus.mercury.*
-import org.hyperledger.identus.mercury.model.error.SendMessageError
 import org.hyperledger.identus.resolvers.DIDResolver
-import org.hyperledger.identus.shared.models.WalletAccessContext
+import org.hyperledger.identus.shared.messaging
+import org.hyperledger.identus.shared.messaging.{Message, WalletIdAndRecordId}
+import org.hyperledger.identus.shared.messaging.MessagingService.RetryStep
+import org.hyperledger.identus.shared.models.{WalletAccessContext, WalletId}
 import org.hyperledger.identus.shared.utils.aspects.CustomMetricsAspect
 import org.hyperledger.identus.shared.utils.DurationOps.toMetricsSeconds
 import zio.*
 import zio.metrics.*
 
+import java.util.UUID
+
 object ConnectBackgroundJobs extends BackgroundJobsHelper {
 
-  val didCommExchanges = {
-    for {
-      connectionService <- ZIO.service[ConnectionService]
-      config <- ZIO.service[AppConfig]
-      records <- connectionService
-        .findRecordsByStatesForAllWallets(
-          ignoreWithZeroRetries = true,
-          limit = config.connect.connectBgJobRecordsLimit,
-          ConnectionRecord.ProtocolState.ConnectionRequestPending,
-          ConnectionRecord.ProtocolState.ConnectionResponsePending
-        )
-      _ <- ZIO.foreachPar(records)(performExchange).withParallelism(config.connect.connectBgJobProcessingParallelism)
-    } yield ()
-  }
+  private val TOPIC_NAME = "connect"
+
+  val connectFlowsHandler = messaging.MessagingService.consumeWithRetryStrategy(
+    "identus-cloud-agent",
+    ConnectBackgroundJobs.handleMessage,
+    Seq(
+      RetryStep(TOPIC_NAME, 5, 0.seconds, s"$TOPIC_NAME-retry-1"),
+      RetryStep(s"$TOPIC_NAME-retry-1", 5, 2.seconds, s"$TOPIC_NAME-retry-2"),
+      RetryStep(s"$TOPIC_NAME-retry-2", 5, 4.seconds, s"$TOPIC_NAME-retry-3"),
+      RetryStep(s"$TOPIC_NAME-retry-3", 5, 8.seconds, s"$TOPIC_NAME-retry-4"),
+      RetryStep(s"$TOPIC_NAME-retry-4", 5, 16.seconds, s"$TOPIC_NAME-DLQ")
```

Comment:
Is there a better solution to keep a single topic with retries?
This solution works, I guess, but it is slightly cumbersome from my point of view.
I would refactor it after merging to the main branch.

Reply:
Not really. Implementing retries with a single queue would lead to "blocking retries" (https://medium.com/@robgolder/kafka-consumer-non-blocking-retry-spring-retry-topics-4ca09675e8b5). Implementing a retry using multiple topics (stateful retry) seems to be a common strategy out there, and the way Spring does it.

Reply:
By the way, in the latest version of this branch, the retry queue list is dynamic and configurable via `application.conf`.